### DISCUSSION TITLE
How to show json_schema_extra in the generated API docs, especially for the "unit" info?

### DISCUSSION THREAD

**Discussion by leiwu84** [fastapi/fastapi#10763] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel, Field


class Apple(BaseModel):
    price: float = Field(
        description="The price of a Fuji apple.",
        json_schema_extra={"unit": "EUR"},
    )
    weight: float = Field(
        description="The weight of a Fuji apple",
        json_schema_extra={"unit": "kg"},
    )


app = FastAPI()


@app.post("")
def run1(input_data: Apple):
    return "Apple"
```

### Description

The "unit" info can be found in `Apple.model_json_schema()`, like `{'price': {'description': 'The price of a Fuji apple.', 'title': 'Price', 'type': 'number', 'unit': 'EUR'},`.

In the generated API docs (http://127.0.0.1:8000/docs), the "unit" info is not shown. 
<img width="302" alt="Screenshot 2023-12-07 at 08 27 15" src="https://github.com/tiangolo/fastapi/assets/68985099/374a3992-01aa-4ac6-b96c-b01809ef3b55">

My users will use the browser API docs to provide input data. It is really useful for them to know the unit of each parameter.

### Operating System

macOS

### Operating System Details

Apple M2 Pro, 2023.

### FastAPI Version

0.104.1

### Pydantic Version

2.5.2

### Python Version

Python 3.11.3

### Additional Context

A similar question is asked on Stackoverflow, https://stackoverflow.com/questions/76904002/how-to-add-unit-in-an-attribute-of-a-class-using-pydantic-basemodel-and-field.
But no answers.

**Comment by YuriiMotov** [comment]
It was fixed and works fine in current version (FastAPI 0.115.12, Pydantic 2.11.5)
